examples/pytorch_lightning/cora_dna.py

```diff
@@ -0,0 +1,370 @@
+"""Graph Convolution Example using Pytorch Geometric
+This example illustrates how one could train a graph convolution model with DNA Conv
+on Cora Dataset using pytorch-lightning. This example will also demonstrate how this
+model can be easily torch-scripted, thanks to Pytorch Geometric.
+"""
+# python imports
+import os
+import os.path as osp
+import sys
+from functools import partial
+from collections import namedtuple
+from argparse import ArgumentParser
+from typing import List, Optional, NamedTuple
+
+# thrid parties libraries
+import numpy as np
+from torch import nn
+import torch
+from torch import Tensor
+from torch.optim import Adam
+import torch.nn.functional as F
+
+# Lightning imports
+from pytorch_lightning import (
+    Trainer,
+    LightningDataModule,
+    LightningModule,
+    seed_everything
+)
+from pytorch_lightning.metrics import Accuracy
+
+try:
+    # Pytorch Geometric imports
+    from torch_geometric.nn import DNAConv, MessagePassing
+    from torch_geometric.data import DataLoader
+    from torch_geometric.datasets import Planetoid
+    import torch_geometric.transforms as T
+    from torch_geometric.data import NeighborSampler
+    from lightning import lightning_logo, nice_print
+except Exception:
+    HAS_PYTORCH_GEOMETRIC = False
+else:
+    HAS_PYTORCH_GEOMETRIC = True
+
+
+# use to make model jittable
+OptTensor = Optional[Tensor]
+ListTensor = List[Tensor]
+
+
+class TensorBatch(NamedTuple):
+    x: Tensor
+    edge_index: ListTensor
+    edge_attr: OptTensor
+    batch: OptTensor
+
+###################################
+#       LightningDataModule       #
+###################################
+
+
+class CoraDataset(LightningDataModule):
+
+    r"""The citation network datasets "Cora", "CiteSeer" and "PubMed" from the
+    `"Revisiting Semi-Supervised Learning with Graph Embeddings"
+    <https://arxiv.org/abs/1603.08861>`_ paper.
+    Nodes represent documents and edges represent citation links.
+    Training, validation and test splits are given by binary masks.
+    c.f https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/datasets/planetoid.py
+    """
+
+    NAME = "cora"
+
+    def __init__(self,
+                 num_workers: int = 1,
+                 batch_size: int = 8,
+                 drop_last: bool = True,
+                 pin_memory: bool = True,
+                 num_layers: int = None):
```

Comment:
```suggestion,
                 num_layers: int
                 num_workers: int = 1,
                 batch_size: int = 8,
                 drop_last: bool = True,
                 pin_memory: bool = True):
```
does not make sense to have it optional if the default value is not feasible...